lib/jekyll-seo-tag/image_drop.rb

```diff
@@ -61,13 +61,17 @@ def raw_path
 
       def absolute_url
         return unless raw_path
-        return @absolute_url if defined? @absolute_url
+        @absolute_url ||= build_absolute_path
+      end
+
+      def build_absolute_path
+        return raw_path unless raw_path.is_a?(String) && absolute_url?(raw_path) == false
+
+        if !raw_path.start_with?("/")
+          return filters.absolute_url File.join(File.dirname(context.registers[:page]["url"]), raw_path)
```

Comment:
Still needs a bit of work, this doesn't work properly for posts with URLs like `http://localhost:4000/post/2022/travel-uke` (with an implied `/index.html`). Good candidate for another unit test.

Reply:
This now works properly with @indirect's changes (thanks!)